### PR TITLE
fix(walletconnect): ensure there is a loading state when preparing transaction

### PR DESCRIPTION
**Pull request by kathaypacific**
### Description

As the title

### Test plan

https://github.com/valora-inc/wallet/assets/20150449/39f0ed1f-dcf0-48f6-a563-bd788fdc5a70

### Related issues

- Fixes RET-971

### Backwards compatibility

Y

### CODE REVIEW

File: src/walletConnect/saga.ts [hunk 1]
```diff
@@ -428,6 +428,13 @@ function* showActionRequest(request: Web3WalletTypes.EventArguments['session_req
     return
   }
 
+  // since there are some network requests needed to prepare the transaction,
+  // add a loading state
+  navigate(Screens.WalletConnectRequest, {
```
Comment (dievazqu) [inline src/walletConnect/saga.ts:433]: Will this create a bottom sheet in top of the other?
For example, if the user is dismissing the "Action" bottom sheet, they will see the loading state?

**Review by jeanregisser — APPROVED** [review]
Awesome! 🚀

File: src/walletConnect/saga.ts (hunk 1 above)
Comment (bakoushin) [reply src/walletConnect/saga.ts:433]: apparently no.
we are already doing [similar trick](https://github.com/valora-inc/wallet/blob/02ba92e639ea3169910896afcc3fd3c61b3d4e2e/src/walletConnect/walletConnect.ts#L66-L87) in session request

**Review by bakoushin — APPROVED** [review]
🎀

**Pull request merged**